### PR TITLE
Kapacitor 1.6.0 jstirnaman

### LINKED ISSUE #2617: Document the `kapacitor/v1/users` API
Document `kapacitor/v1/users` API used to create users in Kapacitor locally. Useful for OSS users given they won't have a meta-node to authenticate against.

Work with @lesam

### PR DESCRIPTION
**Pull request by jstirnaman**
Closes #2617

### CODE REVIEW

**Comment by jstirnaman** [comment]
@sanderson I haven't been able to test this on 1.6.

File: content/kapacitor/v1.6/working/api.md [hunk 1]
```diff
@@ -2250,6 +2251,57 @@ POST /kapacitor/v1/storage/stores/tasks
...
+### Create a user
+To create a user, POST to the `/kapacitor/v1/users` endpoint.
+
+Define a user as a JSON object with the following properties:
+| Property    | Purpose                                                                                   |
+| --------    | -------                                                                                   |
+| name          | The username.                      |
+| password | A password for the user. |
+| type        | The type of user, `user` or `admin`. See [User](/influxdb/v1.6/concepts/glossary/#user).                                                      |
+| permissions | A list of valid user permission strings. Any of `none`, `api`, `config_api`, `write_points`, `all`. |
```
Comment (sanderson) [inline content/kapacitor/v1.6/working/api.md:2274]:
```suggestion
Define a user as a JSON object with the following properties:

| Property    | Description                                                                                |
| :---------- | :----------------------------------------------------------------------------------------- |
| name        | Username                                                                                   |
| password    | Password                                                                                   |
| type        | User type  (`user` or `admin`, _see [User](/influxdb/v1.6/concepts/glossary/#user)_)       |
| permissions | List of valid user permission strings (`none`, `api`, `config_api`, `write_points`, `all`) 
```

File: content/kapacitor/v1.6/working/api.md [hunk 2]
```diff
@@ -2250,6 +2251,57 @@ POST /kapacitor/v1/storage/stores/tasks
...
+
+#### Example
+```sh
+curl -XPOST "http://localhost:9092/kapacitor/v1/users" -d '{ "name": "stan", "password": "pass", "type":"normal", "permissions":["config_api"] }'
+```
```
Comment (sanderson) [inline content/kapacitor/v1.6/working/api.md:2279]:
```suggestion
```sh
curl --XPOST 'http://localhost:9092/kapacitor/v1/users' \
  --data '{
    "name": "stan",
    "password": "pass",
    "type":"normal",
    "permissions": ["config_api"]
}'
```
```

File: content/kapacitor/v1.6/working/api.md [hunk 3]
```diff
@@ -2250,6 +2251,57 @@ POST /kapacitor/v1/storage/stores/tasks
...
+### Update a user
+To update a user, PATCH to the `/kapacitor/v1/users/<name>` endpoint.
+
+Define a modified user as a JSON object with the following properties:
+
+| Property    | Purpose                                                                                   |
+| --------    | -------                                                                                   |
+| password | A password for the user. |
+| type        | The type of user, `user` or `admin`. See [User](/influxdb/v1.6/concepts/glossary/#user).                                                      |
+| permissions | A list of valid user permission strings. Any of `none`, `api`, `config_api`, `write_points`, `all`. |
```
Comment (sanderson) [inline content/kapacitor/v1.6/working/api.md:2290]:
```suggestion
| Property    | Purpose                                                                                    |
| :---------- | :----------------------------------------------------------------------------------------- |
| password    | Password                                                                                   |
| type        | User type (`user` or `admin`, _see [User](/influxdb/v1.6/concepts/glossary/#user)_)        |
| permissions | List of valid user permission strings (`none`, `api`, `config_api`, `write_points`, `all`) |
```

File: content/kapacitor/v1.6/working/api.md [hunk 4]
```diff
@@ -2250,6 +2251,57 @@ POST /kapacitor/v1/storage/stores/tasks
...
+
+#### Example
+```sh
+curl -XPOST "http://localhost:9092/kapacitor/v1/users" -d '{ "name": "stan", "password": "pass", "type":"normal", "permissions":["config_api"] }'
+```
+
+### Update a user
+To update a user, PATCH to the `/kapacitor/v1/users/<name>` endpoint.
```
Comment (sanderson) [inline content/kapacitor/v1.6/working/api.md:2282]:
```suggestion
To update a user, use the PATCH request method with the `/kapacitor/v1/users/<name>` endpoint.
```

File: content/kapacitor/v1.6/working/api.md [hunk 5]
```diff
@@ -2250,6 +2251,57 @@ POST /kapacitor/v1/storage/stores/tasks
...
+### List all users
+To list users, send a GET request to the `/kapacitor/v1/users` endpoint.
+
+#### Example
+```sh
+curl GET "http://localhost:9092/kapacitor/v1/users"
+```
+
+### Create a user
+To create a user, POST to the `/kapacitor/v1/users` endpoint.
```
Comment (sanderson) [inline content/kapacitor/v1.6/working/api.md:2266]:
```suggestion
To create a user, use the POST request method with the `/kapacitor/v1/users` endpoint.
```

File: content/kapacitor/v1.6/working/api.md [hunk 6]
```diff
@@ -2250,6 +2251,57 @@ POST /kapacitor/v1/storage/stores/tasks
 | 400  | Unknown action                     |
 | 404  | The specified store does not exist |
 
+## Users
+Kapacitor exposes operations to manage users.
+
+### List all users
+To list users, send a GET request to the `/kapacitor/v1/users` endpoint.
```
Comment (sanderson) [inline content/kapacitor/v1.6/working/api.md:2258]:
```suggestion
To list users, use the GET request method with the `/kapacitor/v1/users` endpoint.
```

File: content/kapacitor/v1.6/working/api.md [hunk 7]
```diff
@@ -2250,6 +2251,57 @@ POST /kapacitor/v1/storage/stores/tasks
...
+| type        | The type of user, `user` or `admin`. See [User](/influxdb/v1.6/concepts/glossary/#user).                                                      |
+| permissions | A list of valid user permission strings. Any of `none`, `api`, `config_api`, `write_points`, `all`. |
+
+#### Example
+```sh
+curl -XPATCH "http://localhost:9092/kapacitor/v1/users/bob" -d '{ "password": "pass", "type":"admin", "permissions":["all"] }'
+```
+
+### Delete a user
+To delete a user, send a DELETE request to the `/kapacitor/v1/users/<name>` endpoint.
```
Comment (sanderson) [inline content/kapacitor/v1.6/working/api.md:2298]:
```suggestion
To delete a user, use the DELETE request method with the `/kapacitor/v1/users/<name>` endpoint.
```

**Review by sanderson — CHANGES_REQUESTED** [review]
Some suggestions.

File: content/kapacitor/v1.6/working/api.md [hunk 8]
```diff
@@ -2250,6 +2251,63 @@ POST /kapacitor/v1/storage/stores/tasks
...
+### Create a user
+To create a user, use the POST request method with the `/kapacitor/v1/users` endpoint.
+
+Define a user as a JSON object with the following properties:
+
+| Property    | Description                                                                                |
+| :---------- | :----------------------------------------------------------------------------------------- |
+| name        | Username                                                                                   |
+| password    | Password                                                                                   |
+| type        | User type  (`user` or `admin`, _see [User](/influxdb/v1.6/concepts/glossary/#user)_)       |
```
Comment (jstirnaman) [inline content/kapacitor/v1.6/working/api.md:2274]:
```suggestion
| type        | User type  (`normal` or `admin`, _see [User](/influxdb/v1.6/concepts/glossary/#user)_)       |
```

File: content/kapacitor/v1.6/working/api.md [hunk 9]
```diff
@@ -2250,6 +2251,63 @@ POST /kapacitor/v1/storage/stores/tasks
...
+
+### Update a user
+To update a user, use the PATCH request method with the `/kapacitor/v1/users/<name>` endpoint.
+
+Define a modified user as a JSON object with the following properties:
+
+| Property    | Purpose                                                                                    |
+| :---------- | :----------------------------------------------------------------------------------------- |
+| password    | Password                                                                                   |
+| type        | User type (`user` or `admin`, _see [User](/influxdb/v1.6/concepts/glossary/#user)_)        |
```
Comment (jstirnaman) [inline content/kapacitor/v1.6/working/api.md:2295]:
```suggestion
| type        | User type (`normal` or `admin`, _see [User](/influxdb/v1.6/concepts/glossary/#user)_)        |
```

**Pull request merged**